backend/src/main/java/com/redhat/cloud/notifications/oapi/OApiFilter.java

```diff
@@ -0,0 +1,203 @@
+                    }
+                    break;
+                case "paths":
+                    JsonObject pathObject2 = new JsonObject();
+                    JsonObject pathsObjectIn = (JsonObject) entry.getValue();
+                    pathsObjectIn.stream().forEach(pathEntry -> {
+                        String path = pathEntry.getKey();
+
+                        JsonObject pathValue = (JsonObject) pathEntry.getValue();
+                        if (!path.endsWith("openapi.json")) { // Skip the openapi endpoint
+                            JsonObject newPathValue = null;
+                            String mangledPath = mangle(path);
+
+                            if (NOTIFICATIONS.equals(what) && path.startsWith(Constants.API_NOTIFICATIONS_V_1_0)) {
+                                newPathValue = filterPrivateOperation(pathValue, true);
+                            } else if (INTEGRATIONS.equals(what) && path.startsWith(Constants.API_INTEGRATIONS_V_1_0)) {
+                                newPathValue = filterPrivateOperation(pathValue, true);
+                            } else if (PRIVATE.equals(what)) {
+                                newPathValue = filterPrivateOperation(pathValue, false);
+                                mangledPath = path;
+                            } else if (INTERNAL.equals(what) && path.startsWith(Constants.INTERNAL)) {
+                                newPathValue = filterPrivateOperation(pathValue, true);
+                            }
+
+                            if (newPathValue != null) {
+                                pathObject2.put(mangledPath, newPathValue);
+                            }
+                        }
+                    });
+                    root.put("paths", pathObject2);
+                    break;
+                case "info":
+                case "servers":
+                    // Nothing. We handle info and servers below.
+                    break;
+                default:
+                    throw new IllegalStateException("Unknown OpenAPI top-level element " + key);
+            }
+        });
+
+        // Add info section
+        root.put("info", new JsonObject()
+                .put("description", "The API for " + uppify(what))
+                .put("version", "1.0")
+                .put("title", uppify(what)));
+
+        // Add servers section
+        JsonArray serversArray = new JsonArray();
+
+        if (what.equals(NOTIFICATIONS)) {
+            serversArray.add(createServer(true, NOTIFICATIONS));
+            serversArray.add(createServer(false, NOTIFICATIONS));
+        } else if (what.equals(INTEGRATIONS)) {
+            serversArray.add(createServer(true, INTEGRATIONS));
+            serversArray.add(createServer(false, INTEGRATIONS));
+        }
+
+        root.put("servers", serversArray);
+
+        return root;
+    }
+
+    private String uppify(String what) {
+        return what.substring(0, 1).toUpperCase() + what.substring(1);
```

Comment:
Could it be that the variable "what" contains uppercase chars somewhere later? For example like: joseJulio?
If yes, the resultright now would be: JoseJulio and we probably need to change it to:

`what.substring(0, 1).toUpperCase() + what.substring(1).toLowerCase();`